src/content/docs/en/recipes/customizing-output-filenames.mdx

```diff
@@ -0,0 +1,106 @@
+import PackageManagerTabs from '~/components/tabs/PackageManagerTabs.astro';
+
+By default, the `astro build` command outputs your built assets from [your project source](/en/basics/project-structure/#src), like JavaScript and CSS files located in the `src/` directory, into an `_astro` directory with hashed filenames (e.g., `_astro/index.DRf8L97S.js`) which are excellent for long-term caching.
+
+Although it is normally not necessary, you can customise the output file names when needed. For example, this can be helpful if you have scripts with names that might trigger ad blockers (e.g. `ads.js`), or if you want to organize your assets with a particular naming convention. By customizing Rollup output options, you can gain more control over your project's build structure, allowing you to meet specific organizational or deployment requirements.
+
+## Recipe
+
+This recipe configures `vite.build.rollupOptions` to output built assets with the following structure and naming pattern:
+-   JavaScript entry files (e.g., scripts directly associated with your pages or layouts): `dist/js/[name]-[hash].js`
+-   JavaScript code-split chunks (e.g., dynamically imported components or shared modules): `dist/js/chunks/[name]-[hash].js`
+-   Other assets (e.g. CSS, images, fonts): `dist/static/[name]-[hash][extname]` (e.g. `dist/static/styles-a1b2c3d4.css`, `dist/static/logo-e5f6g7h8.svg`)
+
+<Steps>
+
+1.  Add Vite Rollup Output Options.
+
+    Modify your `astro.config.mjs` to include the following `vite.build.rollupOptions.output` configuration. This is where you can define the custom naming patterns for your assets using Rollup's [`entryFileNames`](https://rollupjs.org/configuration-options/#output-entryfilenames), [`chunkFileNames`](https://rollupjs.org/configuration-options/#output-chunkfilenames), and [`assetFileNames`](https://rollupjs.org/configuration-options/#output-assetfilenames):
+
+    ```javascript title="astro.config.mjs" ins
+    import { defineConfig } from 'astro/config';
+
+    export default defineConfig({
+      // ...
+      vite: {
+        build: {
+          rollupOptions: {
+            output: {
+              // path names relative to `outDir`
+              entryFileNames: 'js/[name]-[hash].js',
+              chunkFileNames: 'js/chunks/[name]-[hash].js',
+              assetFileNames: 'static/[name]-[hash][extname]',
+            },
+          },
+        },
+      },
+    });
+    ```
+
+    This example uses the following file name placeholders:
+    *   `[name]`: The original name of the file (without the extension and path).
+    *   `[hash]`: A content-based hash generated for the file, crucial for cache busting. You can also specify a length, e.g., `[hash:8]`. This ensures that when you update an asset, the filename changes, forcing browsers to download the new version instead of serving a stale cached version.
+    *   `[extname]`: The original file extension, including the leading dot (e.g., `.js`, `.css`, `.svg`).
```

Comment:
```suggestion
    *   `[hash]`: A content-based hash generated for the file, crucial for cache busting. You can also specify a length, e.g. `[hash:8]`. This ensures that when you update an asset, the filename changes, forcing browsers to download the new version instead of serving a stale cached version.
    *   `[extname]`: The original file extension, including the leading dot (e.g. `.js`, `.css`, `.svg`).
```